sbol_utilities/component.py

```diff
@@ -36,6 +36,26 @@ def walk_tree(obj: sbol3.TopLevel):
     return {c for c in explored if isinstance(c, sbol3.Component)}
 
 
+def is_dna_part(obj: sbol3.Component) -> bool:
+    """Check if an SBOL Component is a DNA Part, i.e., having type 'SBO:DNA', and exactly 1 sequence property
+
+    :param obj: Sbol3 Document to be checked
+    :return: true if dna part
+    """
+    # must have a type of dna
+    def has_dna_type(component: sbol3.Component) -> bool:
+        return any(tyto.SBO.deoxyribonucleic_acid.is_ancestor_of(type) for type in component.types)
+
+    # there must be atleast 1 SO role, among others
+    def check_roles(component: sbol3.Component) -> bool:
+        return any(tyto.SO.get_term_by_uri(role) for role in component.roles)
+
+    # check all conditions
+    if check_roles(obj) and has_dna_type(obj) and len(obj.sequences) == 1:
+        return True
+    return False
```

Comment:
Notice that you've ended up with "if [statement is True" return True, else return False". When that happens, you can just do "return [statement]"